lsst_dashboard/dataset.py

```diff
@@ -31,69 +38,75 @@ def __init__(self, path, tracts=None, filters=None):
         self.tracts = tracts
         self.filters = filters if filters is not None else []
 
+    def parse_metadata_from_file(self):
+        if self.path.joinpath(METADATA_FILENAME).exists():
+            self.metadata_path = self.path.joinpath(METADATA_FILENAME)
+        else:
+            self.metadata_path = Path(os.environ.get('LSST_META', os.curdir)).joinpath(self.path.name, METADATA_FILENAME)
+
+        with self.metadata_path.open('r') as f:
+            self.metadata = yaml.load(f, Loader=yaml.SafeLoader)
+            self.failures = self.metadata.get('failures', {})
+            if self.tracts is None:
+                self.tracts = list(set(x for v in self.metadata['visits'].values() for x in v.keys()))
+
+    def parse_metadata_from_bulter(self):
+        try:
+            print('-- connect to butler --')
+            self.conn = Butler(str(self.path))
+            self.metadata = self.conn.get('qaDashboard_metadata')
+            self.failures = self.metadata.get('failures', {})
+            if not self.filters:
+                self.filters = list(self.metadata['visits'].keys())
+            if not self.tracts:
+                all_tracts = [list(self.metadata['visits'][filt].keys()) for filt in self.filters]
+                self.tracts = list(set([int(y) for x in all_tracts for y in x]))
+        except:
+            raise
```

Comment:
A good thing to do here would be to catch & print the exception information before proceeding to test data, since often this code may hide some stack problem even if actually on the LSST system